### ISSUE TITLE
Makefile Install Prefix

### ISSUE THREAD

**Issue by TekuConcept** [cisco/openh264#3271]
Would it not be ideal to change the install prefix line in the Makefile from `PREFIX=/usr/local` to `PREFIX?=/usr/local`? That way a developer can specify where to install on a *nix system without needing to manually edit the build-system.

**Comment by fermay** [comment]
actually. you can add " PREFIX=/your_path" in your command to modify the library output directory.

**Comment by TekuConcept** [comment]
> actually. you can add " PREFIX=/your_path" in your command to modify the library output directory.

Unless something had changed or my shell was outdated, this is only possible with the conditional assignment `VARIABLE ?= VALUE`, which is why I had asked the question several years ago. I couldn’t automate the build with a dynamic target output without changing the variable’s assignment flavor in the Makefile.

According to the docs, _”`?=` is called a conditional variable assignment operator, because it only has an effect if the variable is not yet defined.”_ The current `VARIABLE = VALUE` ignores the command line argument and just uses the hard-coded script value instead.

**Comment by fermay** [comment]
I do not think so.  if you do not use `override`, you should not get this result. for example:you can creat a Makefile
`CFLAGS = -fPIC
all: clean
.PHONY:clean
clean:
        @echo $(CFLAGS)
`
using command line to execute `make CFLAGS=NEW` , will print "NEW" not "-fPIC".

**Comment by TekuConcept** [comment]
> I do not think so.

It’s been several years since I raised this issue. Things may have changed, or maybe there was something very simple I overlooked. I no longer have the code on me to see what it was.

All I remember is that I couldn’t set the `PREFIX` value from the command line the same way I could with other makefile libraries, and the reason I found had to do with `=` vs `?=`. When I changed OpenH264’s Makefile to use `?=`, the process worked as expected.

‘Other libraries’ is referring to the various dependencies for a simple network-streaming FFmpeg build. What I ended up doing was just creating a patch file and applying it to this library for my build automation.
